libwallet/assets_manager.go

```diff
@@ -940,46 +943,55 @@ func (mgr *AssetsManager) InitializeDEX(ctx context.Context) {
 	setDEXWalletLoader(mgr.WalletWithID)
 
 	logDir := filepath.Dir(mgr.LogFile())
-	dexcl, err := dexc.Start(ctx, mgr.RootDir(), mgr.GetLanguagePreference(), logDir, mgr.GetLogLevels(), mgr.NetType(), 0 /* TODO: Make configurable */)
+	dexClient, err := dexc.Start(ctx, mgr.RootDir(), mgr.GetLanguagePreference(), logDir, mgr.GetLogLevels(), mgr.NetType(), 0 /* TODO: Make configurable */)
 	if err != nil {
 		log.Errorf("Error starting dex client: %v", err)
 		return
 	}
 
 	mgr.dexcMtx.Lock()
-	mgr.dexc = dexcl
+	mgr.dexc = dexClient
 	mgr.dexcMtx.Unlock()
 
 	go func() {
-		<-mgr.dexc.WaitForShutdown()
+		<-dexClient.WaitForShutdown()
 		mgr.dexcMtx.Lock()
 		mgr.dexc = nil
-		// TODO: Also unregister the custom wallet constructors!
 		mgr.dexcMtx.Unlock()
 	}()
 }
 
 func (mgr *AssetsManager) DeleteDEXData() error {
+	if !mgr.DEXCInitialized() {
+		return nil // nothing to do.
+	}
+
+	mgr.dexcMtx.RLock()
+	dexc := mgr.dexc
+	mgr.dexcMtx.RUnlock()
```

Comment:
I think hold a write-lock till the shutdown completes so no other process gains access to a dexc instance that is about to be shutdown.

Reply:
ok.